index.js

```diff
@@ -75,7 +107,15 @@ function logGraphQLDetails (opts, schema, document, context) {
       body: opts.buildBody(context, currentBody),
       variables: opts.logVariables === true ? currentBody?.variables || null : undefined
     }
-  })
+  }
+
+  const logMessage = opts.buildLogMessage(context)
+
+  if (Array.isArray(logMessage)) {
+    return context.reply.request.log[opts.logLevel](logData, ...logMessage)
+  }
+
+  return context.reply.request.log[opts.logLevel](logData, logMessage)
```

Comment:
new tricky with function.apply(context, argsArray) + concat (handle single value or array)

```suggestion
  const logMessage = opts.buildLogMessage?.(context)

  if (!logMessage) {
    context.reply.request.log[opts.logLevel](logData)
    return
  }

  context.reply.request.log[opts.logLevel].apply(context.reply.request.log, [logData].concat(logMessage))
```

Reply:
I were struggling how to append dynamic parameters to log function execution, handling just one value or more in case of array of pino merging object. I found the spread way, but doesn't work if you haven't an array, so i had to use a guard. The way with `apply` + dynamic array of params seems really better because doesn't make difference!